### ISSUE TITLE
Problem with validation function for TCN and NBEATSx

### ISSUE THREAD

**Issue by hugogobato** [Nixtla/neuralforecast#477]
**Describe the bug**
Support for `validation_epoch_end` has been removed in v2.0.0. `TCN` implements this method. You can use the `on_validation_epoch_end` hook instead. To access outputs, save them in-memory as instance attributes. You can find migration examples in https://github.com/Lightning-AI/lightning/pull/16520.

**To Reproduce**
Just try to use TCN of NBEATSx functions. E.g.:
import numpy as np
import pandas as pd
import pytorch_lightning as pl
import matplotlib.pyplot as plt

from neuralforecast import NeuralForecast
from neuralforecast.models import TCN
from neuralforecast.losses.pytorch import GMM, MQLoss, DistributionLoss
from neuralforecast.utils import AirPassengersPanel, AirPassengersStatic
from neuralforecast.tsdataset import TimeSeriesDataset, TimeSeriesLoader

Y_train_df = AirPassengersPanel[AirPassengersPanel.ds<AirPassengersPanel['ds'].values[-12]] # 132 train
Y_test_df = AirPassengersPanel[AirPassengersPanel.ds>=AirPassengersPanel['ds'].values[-12]].reset_index(drop=True) # 12 test

fcst = NeuralForecast(
    models=[TCN(h=12,
                input_size=-1,
                #loss=DistributionLoss(distribution='Normal', level=[80, 90]),
                loss=GMM(n_components=7, return_params=True, level=[80,90]),
                learning_rate=5e-4,
                kernel_size=2,
                dilations=[1,2,4,8,16],
                encoder_hidden_size=128,
                context_size=10,
                decoder_hidden_size=128,
                decoder_layers=2,
                max_steps=500,
                scaler_type='robust',
                futr_exog_list=['y_[lag12]'],
                hist_exog_list=None,
                stat_exog_list=['airline1'],
                )
    ],
    freq='M'
)
fcst.fit(df=Y_train_df, static_df=AirPassengersStatic)
forecasts = fcst.predict(futr_df=Y_test_df)

**Expected behavior**
The function works.

**Screenshots**
N/A.

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by laurencee9** [comment]
I've got the same issue. Since they support 2.0.0, you can downgrade your torch-lightning version. See [requirements](https://github.com/Nixtla/neuralforecast/blob/main/settings.ini#L18)

**Comment by hugogobato** [comment]
@laurencee9, thank you very much for the reply. But now I am getting this issue

Early stopping conditioned on metric `ptl/val_loss` which is not available. Pass in or modify your `EarlyStopping` callback to use any of the following: `train_loss`, `train_loss_step`, `train_loss_epoch`

Would you know how to solve it?

**Comment by FedericoGarza** [comment]
hey @hugogobato and @laurencee9! Thanks for using neuralforecast. We updated the library to make it compatible with `torch` and `ligthning` `>=2.0.0`. We are about to make an official release to pip. In the meantime, could you try installing the library from the main branch to see if that solves the issue? (`pip install "git+https://github.com/nixtla/neuralforecast.git@main"`)

happy forecasting! :)

**Comment by hugogobato** [comment]
@FedericoGarza, thank you very much for your help. I was wondering whether there is the possibility of seeing the prediction of each stack (or even block) for further interpretiability of the neural networks (like it is showed in the NBEATS paper https://doi.org/10.48550/arXiv.1905.10437). If not, do you think it would be possible to add it?

**Comment by kdgutier** [comment]
Hey @hugogobato,

Following the [NHITS notebook](https://github.com/Nixtla/neuralforecast/blob/main/nbs/models.nhits.ipynb) or [NBEATS notebook](https://github.com/Nixtla/neuralforecast/blob/main/nbs/models.nbeatsx.ipynb), you can generate a trend, seasonality, remainder decomposition.

For reference check 
- Figures 1 and 5 from the NHITS paper https://arxiv.org/abs/2201.12886.
- Figure 3 from the NBEATSx paper https://arxiv.org/abs/2104.05522.

**Comment by hugogobato** [comment]
Hey @kdgutier, thank you very much!
